power_grid_model_c/power_grid_model/include/power_grid_model/container.hpp

```diff
@@ -110,6 +110,19 @@ class Container<RetrievableTypes<GettableTypes...>, StorageableTypes...> {
         assert(is_base<Gettable>[idx_2d.group]);
         return (this->*(func_arr[idx_2d.group]))(idx_2d.pos);
     }
+
+#ifndef NDEBUG
```

Comment:
Deemed inefficient util function and is only used in assert

Reply:
hmmm... i've been thinking about two different approaches:
* doing a reverse search instead (`get_item(idx_2d)->id()`) but then you need to know the type
* doing a `Base` component search (`get_item<Base>(idx_2d)->id()`) but then we need to add `Base` to the `AllComponents`, as well as a `requires` clause to verify that `Base` actually has a member `id()`)
* keeping a lookup of IDs by `idx_2d` (similarly to how we have a lookup of `idx_2d` by `ID`, but that introduces more data, which doesn't make sense if it's currently used in debug mode only

since all of those options are inferior, i guess it's alright to have `NDEBUG` then, but please do use a comment to explain.

in addition, can you please use `std::ranges::find` instead of a for loop

Reply:
Addressed.